File: app/src/main/java/com/refrii/client/ImageDownloadTask.java
package com.refrii.client;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.widget.ImageView;

import java.io.IOException;
import java.io.InputStream;
import java.net.MalformedURLException;
import java.net.URL;

/**
 * Created by yusuke on 2017/09/24.
 */

class ImageDownloadTask extends AsyncTask<String, Void, Bitmap> {

    private ImageView imageView;

    public ImageDownloadTask(ImageView view) {
        imageView = view;
    }

    @Override
    protected Bitmap doInBackground(String... params) {
        Bitmap image;
        try {
            URL imageUrl = new URL(params[0]);
            InputStream imageIs;
            imageIs = imageUrl.openStream();
            image = BitmapFactory.decodeStream(imageIs);
            return image;
        } catch (MalformedURLException e) {
            return null;
        } catch (IOException e) {
            return null;
        }
    }

    @Override
    protected void onPostExecute(Bitmap result) {
        imageView.setImageBitmap(result);
    }
}

File: app/src/main/java/com/refrii/client/ImageDownloadTask.kt
package com.refrii.client

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.AsyncTask
import android.widget.ImageView

import java.io.IOException
import java.io.InputStream
import java.net.MalformedURLException
import java.net.URL

/**
 * Created by yusuke on 2017/09/24.
 */

internal class ImageDownloadTask(private val imageView: ImageView) : AsyncTask<String, Void, Bitmap>() {

    override fun doInBackground(vararg params: String): Bitmap? {
        val image: Bitmap
        try {
            val imageUrl = URL(params[0])
            val imageIs: InputStream
            imageIs = imageUrl.openStream()
            image = BitmapFactory.decodeStream(imageIs)
            return image
        } catch (e: MalformedURLException) {
            return null
        } catch (e: IOException) {
            return null
        }

    }

    override fun onPostExecute(result: Bitmap) {
        imageView.setImageBitmap(result)
    }
}